src/Umbraco.Web.UI.Client/src/views/prevalueeditors/treesource.controller.js

```diff
@@ -118,13 +168,69 @@ angular.module('umbraco')
       });
 		}
 
+    // Dynamic Root specific
+    $scope.$watch("model.value.dynamicRoot", function (newVal, oldVal) {
```

Comment:
Instead of watching `model.value.dynamicRoot.originAlias` it watch `dynamicRoot` so changes to both `dynamicRoot.originAlias` and `dynamicRoot.originKey`.

Alternatively it can have two watch or use watchCollection on the two properties, but I think it is okay to watch `dynamicRoot` as it isn't a deep watch and the object only have a few properties.